vllm/multimodal/utils.py

```diff
@@ -20,14 +23,24 @@ def get_aiohttp_client(cls) -> aiohttp.ClientSession:
             connector = aiohttp.TCPConnector()
             cls.aiohttp_client = aiohttp.ClientSession(timeout=timeout,
                                                        connector=connector)
-
         return cls.aiohttp_client
 
+    @classmethod
+    def is_url_allowed(cls, url: str) -> bool:
+        parsed_url = urlparse(url)
+        if parsed_url.scheme not in ["http", "https"]:
+            return False
+        if parsed_url.hostname not in ALLOWED_DOMAINS:
+            return False
```

Comment:
I'm not sure if we want to build this into the API server especially when a lot of users use it out-of-box and this wouldn't work with most URLs.

AFAIK, all the inference engines that currently support VLMs (TGI, lmdeploy) do not have this check, but maybe we could expose this as an arg for the users to specify their allowed domains?

@simon-mo what's your throught on this?